docs/genai/tutorials/phi3-android.md

```diff
@@ -0,0 +1,118 @@
+    Toast.makeText(MainActivity.this, "Model not loaded yet, please wait...", Toast.LENGTH_SHORT).show();
+    return;
+}
+```
+
+### Prompt Template
+On its own, this model's answers can be very long. To format the AI assistant's answers, you can adjust the prompt template. 
+```java
+String promptQuestion = userMsgEdt.getText().toString();
+String promptQuestion_formatted = "<system>You are a helpful AI assistant. Answer in two paragraphs or less<|end|><|user|>"+promptQuestion+"<|end|>\n<assistant|>";
+Log.i("GenAI: prompt question", promptQuestion_formatted);
+```
+You can also include limits such as a max_length or length_penalty to your liking. 
+```java
+generatorParams.setSearchOption("length_penalty", 1000);
+generatorParams.setSearchOption("max_length", 500);
+```
+NOTE: Including a max_length will cut off the assistant's answer once reaching the maximum number of tokens rather than formatting a complete response.
+
+## Run the App
+
+### Download Android Studio
+You will be using [Android Studio](https://developer.android.com/studio) to run the app.
+
+### Download the App
+Clone the [ONNX Runtime Inference Examples](https://github.com/microsoft/onnxruntime-inference-examples/tree/c29d8edd6d010a2649d69f84f54539f1062d776d) repository.
+
+### Enable Developer Mode on Mobile
+On your Android Mobile device, go to "Settings > About Phone > Software information" and tap the "Build Number" tile repeatedly until you see the message “You are now in developer mode”. In "Developer Options", turn on Wireless or USB debugging.
+
+### Open Project in Android Studio
+Open the Phi-3 mobile app in Android Studio (onnxruntime-inference-examples/mobile/examples/phi-3/android/app).
+
+### Connect Device
+To run the app on a device, follow the instructions from the Running Devices tab on the right side panel. You can connect through Wi-Fi or USB.
+![WiFi Instructions](../../../images/phi3_MobileTutorial_RunDevice.png)
+#### Pair over Wi-Fi
+![WiFi Instructions](../../../images/phi3_MobileTutorial_WiFi.png)
+
+### Manage Devices
+You can manage/change devices and device model through the Device Manager tab on the right side panel.
+![WiFi Instructions](../../../images/phi3_MobileTutorial_DeviceManager.png)
+
+### Downloading the App
+Once your device is connected, run the app by using the play button on the top panel. Downloading all packages will take ~10-15 minutes. If you submit a prompt before downloads are complete, you will encounter an error message. Once completed, the logcat (the cat tab on the bottom left panel) will display an "All downloads complete" message.
+
+![WiFi Instructions](../../../images/phi3_MobileTutorial_Error.png)
+
+### Ask questions
+Now that the app is downloaded, you can start asking questions!
+![Example Prompt 1](../../../images/phi3_MobileTutorial_ex1.png)
+![Example Prompt 2](../../../images/phi3_MobileTutorial_ex2.png)
+![Example Prompt 3](../../../images/phi3_MobileTutorial_ex3.png)
```

Comment:
Might be nice to include some sort of conclusion paragraph, links to additional resources, etc. (e.g., for more information on Phi-3, see ...).